### PR TITLE
Fix some includes and lib dependencies in Interpolation

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Minor things I noticed when reviewing #3555.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/ParallelAlgorithms/Interpolation/CMakeLists.txt [hunk 1]
```diff
@@ -30,21 +30,29 @@ add_dependencies(
...
   PUBLIC
   ApparentHorizons
   DataStructures
-  DgSubcell
   Domain
-  DomainStructure
   ErrorHandling
-  EventsAndTriggers
-  GSL::gsl
+  GeneralRelativitySolutions
```
Comment (markscheel) [inline src/ParallelAlgorithms/Interpolation/CMakeLists.txt:42]: Why isn't GeneralRelativitySolutions in INTERFACE?

Comment (nilsvu) [reply src/ParallelAlgorithms/Interpolation/CMakeLists.txt:42]: The KerrHorizon.cpp includes it through KerrHorizon.hpp, so it's needed to compile the lib (see my comment in your docs PR)

**Review by markscheel — APPROVED** [review]
Looks good. There is a failing clang-tidy test though.

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsdeppe** [comment]
I'm fine ignoring clang-tidy failures since it's not on lines this PR touches :)

**Comment by nilsvu** [comment]
Yea it's unrelated to this PR. I took a stab at it but it's connected to https://github.com/sxs-collaboration/spectre/issues/1866, so I stopped.

**Pull request merged**